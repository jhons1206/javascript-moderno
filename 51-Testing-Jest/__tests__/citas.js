import Citas from '../js/classes/Citas';

describe('Probar la clase de Citas', () => {
    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Cenizo',
            propietario: 'Jeremy',
            telefono: '19895234',
            fecha: '02-03-2023',
            hora: '4:46',
            sintomas: 'Solo duerme',
        };

        citas.agregarCita(citaObj);

        // Pruebas (Testing a la clase)
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Silvestre',
            propietario: 'Jeremy',
            telefono: '19895234',
            fecha: '02-03-2023',
            hora: '4:46',
            sintomas: 'Solo duerme',
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});
