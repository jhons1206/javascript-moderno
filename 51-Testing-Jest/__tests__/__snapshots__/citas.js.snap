// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "02-03-2023",
      "hora": "4:46",
      "id": 1677796001304,
      "mascota": "Silvestre",
      "propietario": "Jeremy",
      "sintomas": "Solo duerme",
      "telefono": "19895234",
    },
  ],
}
`;

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "02-03-2023",
      "hora": "4:46",
      "id": 1677796001304,
      "mascota": "Cenizo",
      "propietario": "Jeremy",
      "sintomas": "Solo duerme",
      "telefono": "19895234",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": [],
}
`;
